src/util/sighash.rs

```diff
@@ -172,7 +197,49 @@ impl<'s> ScriptPath<'s> {
     }
 }
 
-impl<R: Deref<Target=Transaction>> SigHashCache<R> {
+impl From<LegacySigHashType> for SigHashType {
+    fn from(s: LegacySigHashType) -> Self {
+        match s {
+            LegacySigHashType::All => SigHashType::All,
+            LegacySigHashType::None => SigHashType::None,
+            LegacySigHashType::Single => SigHashType::Single,
+            LegacySigHashType::AllPlusAnyoneCanPay => SigHashType::AllPlusAnyoneCanPay,
+            LegacySigHashType::NonePlusAnyoneCanPay => SigHashType::NonePlusAnyoneCanPay,
+            LegacySigHashType::SinglePlusAnyoneCanPay => SigHashType::SinglePlusAnyoneCanPay,
+        }
+    }
+}
+
+impl From<SigHashType> for LegacySigHashType {
+    fn from(s: SigHashType) -> Self {
+        match s {
+            SigHashType::Default => LegacySigHashType::All,
```

Comment:
I don't think we should be doing this. We are changing `0x00` to `0x01`.

Reply:
I removed the from implementation entirely in 993dae0, using LegacySigHashType for the segwits methods doesn't require it anymore